cfgmgr/intfmgr.cpp

```diff
@@ -616,6 +617,35 @@ bool IntfMgr::isIntfStateOk(const string &alias)
     return false;
 }
 
+void IntfMgr::delIpv6LinkLocalNeigh(const string &alias)
+{
+    vector<string> neighEntries;
+
+    SWSS_LOG_INFO("Deleting ipv6 link local neighbors for %s", alias.c_str());
+
+    m_neighTable.getKeys(neighEntries);
+    for (auto neighKey : neighEntries)
+    {
+        if (!neighKey.compare(0, alias.size(), alias.c_str()))
+        {
+            vector<string> keys = tokenize(neighKey, ':', 1);
+            if (keys.size() == 2)
+            {
+                IpAddress ipAddress(keys[1]);
+                if (ipAddress.getAddrScope() == IpAddress::AddrScope::LINK_SCOPE)
+                {
+                    stringstream cmd;
+                    string res;
+
+                    cmd << IP_CMD << " neigh del dev " << keys[0] << " " << keys[1] ;
```

Comment:
why don't we just do a flush of neigh on this interface? why to loop through each of the neighbors especially since this happens during a major config change

Reply:
When we do flushing on a interface, the global ipv6 address also gets deleted along with link-local. For an example  
```
root@sonic:/home/admin# 
root@sonic:/home/admin# ip neigh show
1001::2 dev Ethernet64 lladdr cc:37:ab:17:6c:9a router REACHABLE
fe80::ce37:abff:fe17:6c9a dev Ethernet64 lladdr cc:37:ab:17:6c:9a router REACHABLE
root@sonic:/home/admin# 
root@sonic:/home/admin# sonic-db-cli APPL_DB keys \*NEIGH\*
NEIGH_TABLE:Ethernet64:1001::2
NEIGH_TABLE:Ethernet64:fe80::ce37:abff:fe17:6c9a
root@sonic:/home/admin# 
```

Reply:
agree.. but this operation is not done frequently, and flushing the interface, it can also relearn, right?

Reply:
Looks like this is deleting the link-local neighbor only, given that we will not have many link local neighbors, flushing per interface level may not be required.

Reply:
ok, i was thinking if we could avoid this Neighbor handling and scanning the APP_DB in this context. @AkhileshSamineni, please make sure if the delete operation fails (for some reason the linklocal is not present in kernel while attempting to delete) does not cause process exit

Reply:
@prsunny If the delete operation fails for some reason, it does not cause the process exit.
we see below INFO message only
```
Sep  5 13:00:00.887868 sonic INFO swss#/supervisord: intfmgrd RTNETLINK answers: No such file or directory
```